modules/ROOT/pages/reliability-patterns.adoc

```diff
@@ -0,0 +1,162 @@
+
+* For socket-based connections like HTTP, this means returning an "unsuccessful request" response to the client so that the client can retry the request.
+* For resource-based connections like File or FTP, it means not deleting the file, so that it can be reprocessed.
+
+The application logic flow (the right-hand side of the diagram) delivers the message from the Message Source that uses a transactional connector to the business logic for the application.
+
+== Comparing Endpoints in Reliability Patterns
+
+You can couple the reliable acquisition flow with any type of transactional endpoint. You don't have to use a VM endpoint in a reliability pattern. The following diagram illustrates a reliability pattern where the reliable acquisition flow delivers a message to a JMS endpoint:
+
+image::reliabilitypatternwithjms.png[ReliabilityPatternwithJMS]
+
+The VM connector uses in-memory queues by default, this makes it much faster than other transactional endpoints such as JMS.
+
+The following table lists the advantages and disadvantages of using VM, JMS, and JDBC endpoints in a reliable acquisition flow:
+
+[%header,cols="10a,60a,30a"]
+|===
+|Endpoint |Advantage |Disadvantage
+|VM (standalone Mule)
+|
+* Faster than JMS because everything is in memory and in-process.
+* Provided ready to use with Mule. There is no need to purchase additional software.
+|
+* Data is lost when Mule exits.
+* Non-clustered.
+|File-based VM (standalone Mule)
+|
+* Provided ready to use with Mule. There is no need to purchase additional software.
+|
+* Non-clustered.
+|VM (Mule HA cluster)
+|
+* Faster than JMS because everything is in shared memory.
+* Provided ready to use with Mule. There is no need to purchase additional software.
+|
+* Data is lost when the entire Mule cluster exits.
+|JDBC
+|
+* Depending on resources available, can hold a very large amount of information.
+|
+* Can be complex to use since proper DBMS schemas are required
+|JMS (not enabled for persistence)
+|
+* Allows sharing data with other Mule instances or non-Mule clients.
+|
+* It is slower than VM because it is in another process.
+|JMS (enabled for persistence)
+|
+* It is the most reliable choice because messages are persisted to disk.
+|
+* It is slower than VM because of disk access.
+|===
+
+== Implementing a Reliability Pattern
+
+Here is a code example that shows what a complete reliability pattern would look like, including the reliable acquisition flow and the application logic flow. In this example, the reliable acquisition flow part of the reliability pattern is HTTP-to-VM. Note that this example shows VM file persistency, which does not work on clusters.
```

Comment:
Should that be a [NOTE]? Of not, remove "Note" and just Say "This example shows..."